### ISSUE TITLE
Misterious `RangeError: Invalid count value: -1` error

### ISSUE THREAD

**Issue by ilenia-magoni** [freeCodeCamp/freeCodeCamp#55104]
On step 89 and 90 of the Pyramid generator there is this misterious error that appear, it can be confusing for campers.

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-introductory-javascript-by-building-a-pyramid-generator/step-90

![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/26656284/db7cd877-34d5-4435-b465-aad657d9ae46)

**Comment by ojeytonwilliams** [comment]
Yep. I think reworking the lessons so we never leave the learner with broken code (or at least never unexplained broken code) is the way to go.

**Comment by gikf** [comment]
I'm not sure if it was even intended to have broken code, description is suggesting everything is working fine. Step 91 asks for two changes, one of which specifically makes this error appear (second change fixes the issue).

**Comment by ojeytonwilliams** [comment]
Yeah, it's a bit confusing. Probably not intentional, though. On 91 there's this:

> When you do this, you may see a Range Error, because we've created another off-by-one error.

...but the only time I _don't_ see the error is when I've followed all the instructions on step 91.

**Comment by Ksound22** [comment]
> Yeah, it's a bit confusing. Probably not intentional, though. On 91 there's this:
> 
> > When you do this, you may see a Range Error, because we've created another off-by-one error.
> 
> ...but the only time I _don't_ see the error is when I've followed all the instructions on step 91.

Is this not enough to clear up any doubts? Cuz after following the instructions in step 91, there's nothing mysterious about the error anymore.

**Comment by ojeytonwilliams** [comment]
I don't think so.

Step 89, where error is introduced, doesn't mention it. Neither does 90. It's only when you reach 91 that the error is mentioned and the text is 

> So you'll need to replace done here with rows.length + 1. When you do this, you may see a Range Error, because we've created another off-by-one error.

which is incorrect. Changing `done` to `rows.length + 1` has no effect on the error. It's only the combination of that and adjusting the conditional that finally removes the error.

The main issue with this is that it breaks the feedback loop between learner's actions and the code's behaviour. Speaking for myself, everything gets harder when the results of my actions are essentially invisible. I know we run tests to check you re doing it right, but that's not really a substitute for them seeing how your actions impact the program.

**Comment by a2937** [comment]
So we should mention the error on Step 90 then?

**Comment by naomi-lgbt** [comment]
Just for posterity, the steps got shifted.

- Error is introduced by completing step 95
- Error is resolved by completing step 96

What should be changed here:

- Step 95 should:
  - Introduce and explain the error
  - Ask the campers to change the `<=` comparison to `<`
- Step 96 should:
  - No longer ask campers to change the `<=` to `<`
  - Not mention that `rows.length + 1` creates a `RangeError` ||it doesn't||
